src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1230,7 +1231,6 @@ private void retrieveChanges(Run build, GitClient git, TaskListener listener) th
                     removeSecondFetch = determineSecondFetch(option, rc);
                 }
             } catch (GitException ex) {
-                ex.printStackTrace(listener.error("Error cloning remote repo '" + rc.getName() + "'"));
```

Comment:
> > another test we did is to build the git client using OpenSSL instead of GnuTLS but we still got issues like:
> > OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to github.com:443
> 
> Command line git does not rely on the SSL implementation inside the git client plugin. Command line git uses its own implementation. If you're not running the most recent release of command line git, you might consider trying it, since there may be improvements in recent releases that help your case.

This error related to git which built with OpenSSL instead of GnuTLS , it is a test we did to verify that the issues we are facing are not related to GnuTLS, we already reverted to GnuTLS as we saw that Git with OpenSSL doesn't solve the issue.

We've updated the Git Client to the latest version on both of our agents (k8s pod - version jenkins/inbound-agent:3107.v665000b_51092-15-jdk11, and VM <debian11> using the Azure VM Agent plugin)

We've also tried to change the git configuration as suggested in few stackoverflow issues but it didn't helped (like :
`	git config --system core.compression 0
	git config --system http.postBuffer 1048576000
	git config --system http.maxRequestBuffer 100M
	git config --system https.postBuffer 1048576000
	git config --system https.maxRequestBuffer 100M`
)
Another thing we've tried is to increase the git clone timeout by:
`jvmOptions: "-Dorg.jenkinsci.plugins.gitclient.Git.timeOut=30"
`
as I mentioned above, we are running 1000+ jobs in parallel which triggers huge amount of pods in AKS and VM (using the Azure VM Agent plugin) which result to 10-30~ of jobs to fail on the initial git clone on the agent (after fetching the Jenkins file in master) 

adding more errors examples:

error 1:
`16:06:44  stderr: error: RPC failed; HTTP 408 curl 22 The requested URL returned error: 408
16:06:44  fatal: expected 'packfile'`

error 2:
`gnutls_handshake() failed: The TLS connection was non-properly terminated.`

error 3:
`Receiving objects:  80% (236132/292087), 470.98 MiB | 97.00 KiB/s
Receiving objects:  80% (236132/292087), 471.04 MiB | 95.00 KiB/s
Receiving objects:  80% (236132/292087), 471.17 MiB | 95.00 KiB/s
fetch-pack: unexpected disconnect while reading sideband packet
17:09:22  fatal: early EOF
17:09:22  fatal: fetch-pack: invalid index-pack output`

Jenkins version: [Jenkins 2.401.1]

Reply:
You are fetching a 470 MB repository hundreds of times in parallel.  A reference repository will save most of that data transfer and will save much computation on the git server.

Reply:
@MarkEWaite Since we are using this patch our issues were solved.
What is your thoughts of merging this PR?

Reply:
@itzikbekelmicrosoft I don't plan to merge this pull  request.  It ignores the solution that is already available (use a reference repository to reduce data transfer and load on the central repository) and changes existing behavior for the many existing users of the plugin.